Framework/CurveFitting/src/Algorithms/SplineInterpolation.cpp

```diff
@@ -253,6 +334,88 @@ void SplineInterpolation::calculateSpline(
   m_cspline->function1D(yValues, xValues, nData);
 }
 
+/** Extrapolates flat for the points outside the x-range
+ * This is used for linear case only, to be consistent with cubic spline case
+ * @param ows : output workspace
+ * @param iwspt : workspace to interpolate
+ * @param row : the workspace index
+ * @param indices : the pair of x-axis indices defining the extrapolation range
+ * @param doDerivs : whether derivatives are requested
+ * @param derivs : the vector of derivative workspaces
+*/
+void SplineInterpolation::extrapolateFlat(
+    MatrixWorkspace_sptr ows, MatrixWorkspace_const_sptr iwspt,
+    const size_t row, const std::pair<size_t, size_t> &indices,
+    const bool doDerivs, std::vector<MatrixWorkspace_sptr> &derivs) const {
+
+  const double yFirst = iwspt->y(row).front();
+  const double yLast = iwspt->y(row).back();
+  for (size_t bin = 0; bin < indices.first; ++bin) {
+    ows->mutableY(row)[bin] = yFirst;
+    if (doDerivs) {
+      // if derivatives are requested
+      derivs[row]->mutableY(0)[bin] = 0.;
+    }
+  }
+  for (size_t bin = indices.second; bin < ows->blocksize(); ++bin) {
+    ows->mutableY(row)[bin] = yLast;
+    if (doDerivs) {
+      // if derivatives are requested
+      derivs[row]->mutableY(0)[bin] = 0.;
+    }
+  }
+}
+
+/** Find the region that has to be interpolated
+ * E.g. iwspt x-axis is from 50-100, while mwspt x-axis is 0-200, this will
+ * return
+ * the pair of the indices of mwspt, that are just above 50, and just below 100
+ * This is used for linear case only, to be consistent with cubic spline case
+ * @param iwspt : workspace to interpolate
+ * @param mwspt : workspace to match
+ * @param row : the workspace index
+ * @return : pair of indices for representing the interpolation range
+*/
+std::pair<size_t, size_t>
+SplineInterpolation::findInterpolationRange(MatrixWorkspace_const_sptr iwspt,
+                                            MatrixWorkspace_sptr mwspt,
+                                            const size_t row) {
+
+  auto xAxisIn = iwspt->x(row).rawData();
+  std::sort(xAxisIn.begin(), xAxisIn.end());
+  const auto &xAxisOut = mwspt->x(0).rawData();
+
+  size_t firstIndex = 0;
+  size_t lastIndex = xAxisOut.size();
+
+  if (xAxisOut.front() >= xAxisIn.back()) {
+    lastIndex = firstIndex;
+  } else if (xAxisOut.back() <= xAxisIn.front()) {
+    firstIndex = lastIndex;
+  } else {
+    for (size_t i = 0; i < xAxisOut.size(); ++i) {
+      if (xAxisOut[i] > xAxisIn.front()) {
+        firstIndex = i;
+        break;
+      }
+    }
+    for (size_t i = 0; i < xAxisOut.size(); ++i) {
+      if (xAxisOut[i] > xAxisIn.back()) {
+        lastIndex = i;
+        break;
+      }
+    }
+  }
+
+  std::stringstream log;
+  log << "Workspace index " << row
+      << ": Will perform flat extrapolation outside bin range: " << firstIndex
+      << " to " << lastIndex << "\n";
```

Comment:
You can avoid `stringstream` by using [`std::to_string`](http://en.cppreference.com/w/cpp/string/basic_string/to_string)